core/connector/protobuf_configmap.go

```diff
@@ -0,0 +1,170 @@
+)
+
+// ProtoBufConfigMapHandler ...
+type ProtoBufConfigMapHandler interface {
+	GetProtoBufConfigMapName(runtimeInstance api.KogitoRuntimeInterface) string
+	CreateProtoBufConfigMap(runtimeInstance api.KogitoRuntimeInterface) *corev1.ConfigMap
+	FetchProtoBufConfigMap(runtimeInstance api.KogitoRuntimeInterface) (*corev1.ConfigMap, error)
+}
+
+type protobufConfigMapHandler struct {
+	operator.Context
+	deploymentHandler    kogitoservice.DeploymentHandler
+	kogitoServiceHandler kogitoservice.ServiceHandler
+	configMapHandler     infrastructure.ConfigMapHandler
+}
+
+// NewProtoBufConfigMapHandler ...
+func NewProtoBufConfigMapHandler(context operator.Context) ProtoBufConfigMapHandler {
+	return &protobufConfigMapHandler{
+		Context:              context,
+		deploymentHandler:    kogitoservice.NewDeploymentHandler(context),
+		kogitoServiceHandler: kogitoservice.NewKogitoServiceHandler(context),
+		configMapHandler:     infrastructure.NewConfigMapHandler(context),
+	}
+}
+
+func (p *protobufConfigMapHandler) FetchProtoBufConfigMap(runtimeInstance api.KogitoRuntimeInterface) (*corev1.ConfigMap, error) {
+	labelSelector := map[string]string{
+		framework.LabelAppKey:            runtimeInstance.GetName(),
+		ConfigMapProtoBufEnabledLabelKey: "true",
+	}
+	configMapList, err := p.configMapHandler.FetchConfigMapsForLabel(runtimeInstance.GetNamespace(), labelSelector)
+	if err != nil {
+		return nil, err
+	}
+	if len(configMapList.Items) > 0 {
+		return &configMapList.Items[0], nil
+	}
+	return nil, nil
+}
+
+func (p *protobufConfigMapHandler) CreateProtoBufConfigMap(runtimeInstance api.KogitoRuntimeInterface) *corev1.ConfigMap {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: runtimeInstance.GetNamespace(),
+			Name:      p.GetProtoBufConfigMapName(runtimeInstance),
+			Labels: map[string]string{
+				ConfigMapProtoBufEnabledLabelKey: "true",
+				framework.LabelAppKey:            runtimeInstance.GetName(),
+			},
+			Annotations: map[string]string{
+				infrastructure.FromFileKey:  "true",
+				infrastructure.MountPathKey: path.Join(DefaultProtobufMountPath, runtimeInstance.GetName()),
+				infrastructure.FileModeKey:  fmt.Sprint(framework.ModeForProtoBufConfigMapVolume),
+			},
+		},
+		Data: p.getProtobufData(runtimeInstance),
+	}
+	return configMap
+}
+
+func (p *protobufConfigMapHandler) getProtobufData(runtimeInstance api.KogitoRuntimeInterface) map[string]string {
+	available, err := p.deploymentHandler.IsDeploymentAvailable(runtimeInstance)
+	if err != nil {
+		p.Log.Error(err, "failed to check deployment status")
+		return nil
+	}
+	if !available {
+		p.Log.Debug("deployment not available")
+		return nil
+	}
+
+	protobufEndpoint := p.kogitoServiceHandler.GetKogitoServiceEndpoint(runtimeInstance) + protobufSubdir
+	protobufListURL := protobufEndpoint + protobufListFileName
+	protobufListBytes, err := getHTTPFileBytes(protobufListURL)
+	if err != nil {
+		p.Log.Error(err, "failed to get protobuf file list")
```

Comment:
append the file/url here or will it be printed on the err ?

Reply:
done